### PR TITLE
agate: update `encode` and `decode` for `Value`

### PR DESCRIPTION
**Pull request by GanZiheng**
Signed-off-by: GanZiheng <ganziheng98@gmail.com>

### CODE REVIEW

File: src/value.rs [hunk 1]
```diff
@@ -211,3 +211,21 @@ impl EntryReader {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_encode_decode_var() {
```
Comment (Connor1996) [inline src/value.rs:220]: add check for `encode` and `decode` as well

File: src/value.rs [hunk 2]
```diff
@@ -117,21 +117,19 @@ impl Value {
...
+        self.expires_at = decode_length_delimiter(&mut bytes).unwrap() as u64;
+        self.value = bytes.to_owned();
     }
 
     pub fn encode(&self, buf: &mut BytesMut) {
-        let arr: [MaybeUninit<u8>; 12] = unsafe { MaybeUninit::uninit().assume_init() };
-        let mut arr = unsafe { std::mem::transmute::<_, [u8; 12]>(arr) };
-        arr[0] = self.meta;
-        arr[1] = self.user_meta;
-        let written = encode_var(&mut arr[2..], self.expires_at);
```
Comment (Connor1996) [inline src/value.rs:133]: seems we can delete `encode_var` and `decode_var` now

File: src/value.rs [hunk 3]
```diff
@@ -212,3 +148,23 @@ impl EntryReader {
...
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_encode_decode() {
+        let v = Value::new(Bytes::from("hello world"));
```
Comment (tabokie) [inline src/value.rs:158]: Test `new_with_meta`

**Review by tabokie — APPROVED** [review]

File: src/value.rs [hunk 4]
```diff
@@ -109,29 +49,25 @@ impl Value {
...
+            + self.value.len()
     }
 
-    pub fn decode(&mut self, bytes: &Bytes) {
-        self.meta = bytes[0];
-        self.user_meta = bytes[1];
-        let res = decode_var(&bytes[2..]);
-        self.expires_at = res.0;
-        self.value = bytes.slice(res.1 + 2..);
+    pub fn decode(&mut self, mut bytes: &mut Bytes) {
```
Comment (Connor1996) [inline src/value.rs:58]:
```suggestion
    pub fn decode(&mut self, mut bytes: Bytes) {
```

File: src/value.rs [hunk 5]
```diff
@@ -109,29 +49,25 @@ impl Value {
...
+        self.meta = bytes.get_u8();
+        self.user_meta = bytes.get_u8();
+
+        self.expires_at = decode_length_delimiter(&mut bytes).unwrap() as u64;
+        self.value = bytes.to_owned();
```
Comment (Connor1996) [inline src/value.rs:63]:
```suggestion
        self.value = bytes;
```

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**